WordPress/Classes/ViewRelated/Media/Tenor/TenorResponse.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+import MobileCoreServices
+
+typealias TenorGifFormat = TenorMedia.MediaCodingKeys
+
+class TenorResponse: NSObject, Decodable {
+    let next: String
+    let results: [TenorMedia]
+}
+
+class TenorMedia: NSObject, Decodable {
+    enum CodingKeys: String, CodingKey {
+        case id
+        case created
+        case itemurl
+        case title
+        case media
+    }
+
+    enum MediaCodingKeys: String, CodingKey, CaseIterable {
+        case nanogif
+        case tinygif
+        case mediumgif
+        case gif
+    }
+
+    let id: String
+    let created: Date
+    let itemurl: String
+    let title: String
+
+    // Data format in the response has unnecessary depth. We'll convert it to a more optimal format
+    var gifs: [TenorGifFormat: TenorGif]
+
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        id = try container.decode(String.self, forKey: CodingKeys.id)
+
+        let createdDate = try container.decode(Float.self, forKey: CodingKeys.created)
+        created = Date(timeIntervalSince1970: Double(createdDate))
+
+        itemurl = try container.decode(String.self, forKey: CodingKeys.itemurl)
+        title = try container.decode(String.self, forKey: CodingKeys.title)
+
+        var mediaContainer = try container.nestedUnkeyedContainer(forKey: CodingKeys.media)
+
+        gifs = [TenorGifFormat: TenorGif]()
+
+        while !mediaContainer.isAtEnd {
+            let gifsContainer = try mediaContainer.nestedContainer(keyedBy: MediaCodingKeys.self)
+            for key in MediaCodingKeys.allCases {
+                gifs[key] = try gifsContainer.decode(TenorGif.self, forKey: key)
+            }
+        }
+    }
+}
+
+class TenorGif: NSObject, Codable {
+    let url: String
+    let dims: [Int]
+    let preview: String
+}
+
+// MARK: - Helpers needed by WPMediaAsset conformance
+
+extension TenorMedia {
+    var previewGif: TenorGif {
+        return gifs[.tinygif]!
+    }
+
+    var largeGif: TenorGif {
+        return gifs[.gif]!
+    }
+
+    var previewURL: URL {
+        return NSURL(string: previewGif.url)! as URL
+    }
+
+    var staticThumbnailURL: URL {
+        return NSURL(string: previewGif.preview)! as URL
```

Comment:
I would try to avoid using force unwraps. Can we make these properties Optionals?

Reply:
You're right. In normal circumstances these should be safe unwraps, because each item in the response should always have these fields. But we shouldn't rely on the integrity of the data coming from a third party, therefore eliminating the IUOs would be the best course of action. The optionals will propagate deeper into the code, but better safe than sorry.